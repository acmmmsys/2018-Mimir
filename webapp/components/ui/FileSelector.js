import React, { Component } from 'react';
import { connect } from 'react-redux';
import classnames from 'classnames';

import Image from 'components/ui/Image';

import 'style/ui/FileSelector.scss';

const defaultClasses = [
    'image-container',
];

const sortFiles = (a, b) => {
    return (a.id < b.id) ? -1 : (a.id > b.id) ? 1 : 0;
}

class FileSelector extends Component {

    static defaultProps = {
        classNames: [],
        files: [],
    }
    
    getFileArray = () => {
        return Object.keys(this.props.files).
            map(fileId => this.props.files[fileId]).sort(sortFiles);
    }
    
    render() {

        let classNames = classnames(
            defaultClasses,
            this.props.classNames,
        );

        return (
            <div className={classNames}>
                {this.getFileArray().map(file => (
                    <Image
                        width={100}
                        height={100}
                        key={file.id} 
                        selected={file.id === this.props.selectedFile}
                        loading={file.loading}
                        attached={file.attached}
                        label={file.type}
                        clickable={true}
                        src={file.source} 
                        onClick={() => this.props.selectFile(file.id)}
                    />
                ))}
            </div>
        );
    }
}

export default FileSelector;