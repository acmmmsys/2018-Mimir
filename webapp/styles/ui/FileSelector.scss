@import "~compass-mixins/lib/compass";

.image-container {
    width: 100%;
    height: 100%;
    display: grid;
    overflow-y: scroll;
    grid-gap: 10px;
    box-sizing: border-box;
    grid-template-columns: repeat(auto-fill, minmax(105px, 1fr));
    grid-template-rows: repeat(auto-fill, minmax(105px, 1fr));
    min-height: 0;
    min-width: 0; 
    justify-items: center;
    align-items: center;
}