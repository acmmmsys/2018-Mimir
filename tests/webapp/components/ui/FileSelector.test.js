import React from 'react';
import FileSelector from '../../../../webapp/components/ui/FileSelector'
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store'
 
const mockStore = configureStore();

let store;
let wrapper;

test('FileSelector does render', () => {
  const wrapper = shallow(
      <FileSelector store={mockStore(initialState)}>This is a test!</FileSelector>
  );
  expect(wrapper).toMatchSnapshot();
});
