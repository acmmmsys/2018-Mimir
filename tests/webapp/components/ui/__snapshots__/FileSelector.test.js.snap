// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileSelector does render 1`] = `
<div
  className="image-container "
/>
`;
